publish.gradle

```diff
@@ -185,25 +185,47 @@ ext.configurePublish = { jnaForTestConfiguration = null ->
 
 
 
-// A convenience function for configuring a `uniffi-bindgen` task,
-// with appropriate dependency info. This fill call `uniffi-bindgen`
-// on the provided `.udl` file in order to generate Kotlin language
+// A convenience function for configuring a `uniffi-bindgen` task, with appropriate dependency info.
+// This will call `uniffi-bindgen` with the provided crate in order to generate Kotlin language
 // bindings and include them in the source set for the project.
-ext.configureUniFFIBindgen = { udlFilePath ->
+ext.configureUniFFIBindgen = { crateName ->
     android.libraryVariants.all { variant ->
         def uniffiGeneratedPath = "generated/source/uniffi/${variant.name}/java"
-        def t = tasks.register("generate${variant.name.capitalize()}UniFFIBindings", Exec) {
-            workingDir project.rootDir
-            commandLine '/usr/bin/env', 'cargo', 'uniffi-bindgen', 'generate', "${project.projectDir}/${udlFilePath}", '--language', 'kotlin', '--out-dir', "${buildDir}/${uniffiGeneratedPath}"
+        def megazordPathCandidates = [
+            "${project.rootDir}/target/release/libmegazord.so",
+            "${project.rootDir}/target/release/libmegazord.dylib",
+            "${project.rootDir}/target/release/libmegazord.dll"
+        ]
+        def taskName = "generate${variant.name.capitalize()}UniFFIBindings";
+        def t = tasks.register(taskName) {
+            doLast {
+                def megazordPath = null
+                for (path in megazordPathCandidates) {
+                    def file = new File(path)
+                    if (file.exists()) {
+                        megazordPath = path
+                        break
+                    }
+                }
+                if (megazordPath == null) {
+                    throw new GradleException("libmegazord dynamic library path not found")
+                }
+                exec {
+                    workingDir project.rootDir
+                    commandLine '/usr/bin/env', 'cargo', 'uniffi-bindgen', 'generate', '--library', megazordPath, "--crate", crateName, '--language', 'kotlin', '--out-dir', "${buildDir}/${uniffiGeneratedPath}"
+                }
+            }
             outputs.dir "${buildDir}/${uniffiGeneratedPath}"
-            // Re-generate if the interface definition changes.
-            inputs.file "${project.projectDir}/${udlFilePath}"
+            // Re-generate when the megazord is rebuilt, which should happen if any components
+            // change
+            it.inputs.files megazordPathCandidates
             // Re-generate if our uniffi-bindgen tooling changes.
             inputs.dir "${project.rootDir}/tools/embedded-uniffi-bindgen/"
             // Re-generate if our uniffi-bindgen version changes.
             inputs.file "${project.rootDir}/Cargo.lock"
         }
         variant.registerJavaGeneratingTask(t.get(), new File(buildDir, uniffiGeneratedPath))
+        tasks[taskName].dependsOn(":full-megazord:assembleRelease")
```

Comment:
`t.get()` (and `tasks[taskName]`) will force evaluation of the task graph—which can have unintended consequences down the line—and there's new lazy configuration APIs that we can use here, for both Gradle and the Android Gradle plug-in. Not at all a blocker for this PR, since we were already doing this, but I can get a follow-up bug on file for that!